File: components/jpa-handling-session/src/main/java/com/sv/youapp/component/authorization/repositories/jpa/ClientRepository.java
package com.sv.youapp.component.authorization.repositories.jpa;

import com.sv.youapp.component.authorization.entities.jpa.RegisteredClientEntity;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface ClientRepository extends JpaRepository<RegisteredClientEntity, String> {

    @EntityGraph(attributePaths = {
        "authenticationMethods",
        "grantTypes",
        "scopes",
        "redirectUris",
        "postLogoutRedirectUris",
        "clientSettings",
        "tokenSettings"
    })
    Optional<RegisteredClientEntity> findByClientId(String clientId);

	@EntityGraph(attributePaths = {
        "authenticationMethods",
        "grantTypes",
        "scopes",
        "redirectUris",
        "postLogoutRedirectUris",
        "clientSettings",
        "tokenSettings"
    })
    Optional<RegisteredClientEntity> findById(String id);
}

File: components/jpa-handling-session/src/main/kotlin/com/sv/youapp/component/authorization/repositories/jpa/ClientRepository.kt
package com.sv.youapp.component.authorization.repositories.jpa

import com.sv.youapp.component.authorization.entities.jpa.RegisteredClientEntity
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface ClientRepository : JpaRepository<RegisteredClientEntity, String> {
    @EntityGraph(
        attributePaths = [
            "authenticationMethods",
            "grantTypes",
            "scopes",
            "redirectUris",
            "postLogoutRedirectUris",
            "clientSettings",
            "tokenSettings",
        ],
    )
    fun findByClientId(clientId: String): Optional<RegisteredClientEntity>

    @EntityGraph(
        attributePaths = [
            "authenticationMethods",
            "grantTypes",
            "scopes",
            "redirectUris",
            "postLogoutRedirectUris",
            "clientSettings",
            "tokenSettings",
        ],
    )
    override fun findById(id: String): Optional<RegisteredClientEntity>
}
